src/utils/byteorder.h.in

```diff
@@ -65,10 +65,14 @@
 # include "integers.h"
 #endif
 
-
 #ifdef HAVE_ENDIAN_H
 # include <endian.h>
-#else
+#endif
+
+#ifdef HAVE_BYTESWAP_H
```

Comment:
Just a general comment: A lot of different defines makes for multiple code paths that must be tested. I suggest reviewing these and identifying if any of the #defines can be removed. It will make the code simpler and easier to test.

Reply:
It was actually much worse before. Unless we only will use our own implementation of these functions (ignoring system-wise implementations even if they exists), I think it is difficult to simplify this further.